lib-sql/functions/place_triggers.sql

```diff
@@ -77,6 +77,12 @@ BEGIN
 
   ELSE -- insert to placex
 
+    -- Pure postcodes are never queried from placex so we don't add them.
+    -- location_postcodes is filled from the place table directly.
+    IF NEW.class = 'place' AND NEW.type = 'postcode' THEN
+      RETURN NEW;
```

Comment:
We have to handle a type change here. For example if an OSM object was tagged `place=town,addr:postcode=123` and somebody deletes the `place` tag, then the final type changes from `town` to `placecode` and you need to make sure to remove the old town entry from placex. The same delete as in line 105 should do.

We really need to add tests to the BDD test suite for these kind of corner cases.

Reply:
Concerning the BDD tests, I need to have a deeper look on how to make them right. As I try to focus on my priorities for the GSoC, is it ok to have another PR for those tests later? (Of course I will finish to correct this PR till it's ok to be merged)

Reply:
I agree. Let's leave that for a later PR.